amosproj/amos2021ws07-nft-development
# 97 buy nft button and page to interact with contract

This PR implements a page to interact with the most recent WIP NFT drop contract. In particular, the user can create a new drop, join a drop, drop a drop, buy nfts that were assigned to a user and show the NFT index of owned NFTs. This PR only implements the logic and no good design.

Screenshots:

![image](https://user-images.githubusercontent.com/58464184/144185178-91aa2d11-167d-4294-9ddc-cd1a481c53e2.png)
![image](https://user-images.githubusercontent.com/58464184/144185199-57bd4053-9732-480f-a193-cd4874cc75a8.png)

How to test:
0.  Have MetaMask installed and have some Kovan ETH.
1. Create a new drop: specify number of NFTs to create (e.g. 100) and droptime (e.g. 0).  Confirm in MetaMask.
2. Go to https://kovan.etherscan.io/address/0xA8C4e9a4B7F6d5A818b5f2d6D7FB6d202D542646#readContract and execute the `dropHashes` function with the new ID of the drop. At the moment of this PR, there are already 4 drops (ID 0,1,2,3). Thus, any new contract should have a higher ID. Try to find the highest ID.
3. Enter the drop hash number received from the `dropHashes` function into the "Drop hash number" field, enter a number of NFTs to join drop for and click "join drop". Confirm in MetaMask.
4. Click the drop drop button. Confirm in MetaMask.
5. Go back to https://kovan.etherscan.io/address/0xA8C4e9a4B7F6d5A818b5f2d6D7FB6d202D542646#readContract and run the `nftAssetsInformationOfUsers` function. Enter your address used in the frontend and in the second field a number between 0 and the number of NFTs you joined for in the drop minus one. I.e. if you joined for 5 NFTs, enter a number between 0 and 4. 
6. Copy the number into the NFT hash to buy field and click "Buy NFT". Confirm in MetaMask. You can repeat (5) and (6) as often as you joined the drop.
7. In the "Get index (0,1,2...) of owned NFT" field, enter a number between 0 and the number of bough NFTs minus 1. Click the "Get real NFT index of owned NFTs". The real NFT hash of the NFT should be displayed.
